components/content_settings/core/browser/brave_content_settings_pref_provider.cc

```diff
@@ -425,6 +433,42 @@ void BravePrefProvider::MigrateShieldsSettingsV1ToV2ForOneType(
   }
 }
 
+void BravePrefProvider::MigrateFPShieldsSettingsAndroid() {
+  if (prefs_->GetBoolean(kBraveShieldsFPSettingsMigration))
+    return;
+
+  // Find rules that can be migrated and create replacement rules for them.
+  std::vector<Rule> rules;
+  auto rule_iterator = PrefProvider::GetRuleIterator(
+      ContentSettingsType::BRAVE_FINGERPRINTING_V2,
+      /*off_the_record*/ false);
+  while (rule_iterator && rule_iterator->HasNext()) {
+    auto rule = rule_iterator->Next();
+    rules.emplace_back(CloneRule(rule));
+  }
+  rule_iterator.reset();
+
+  // Migrate.
+  for (const auto& fp_rule : rules) {
+    if (fp_rule.secondary_pattern == ContentSettingsPattern::Wildcard() &&
+        fp_rule.value == CONTENT_SETTING_BLOCK) {
+      SetWebsiteSettingInternal(fp_rule.primary_pattern,
```

Comment:
If sync is enabled then these rules will be returned back in few seconds after migration. Because semantic is changed you should change path in the prefs. 

1. change value of kFingerprintingV2 (brave_shield_constants.h) to "fingerprintingV3"
2. register "profile.content_settings.exceptions.fingerprintingV2" in BravePrefProvider::RegisterProfilePrefs
3. In MigrateFPShieldsSettingsAndroid copy all you need from fingerprintingV2 into fingerprintingV3
4. Do migration for desktop as well because if sync is enabled the desktop and mobile clients should understand these settings identically

Reply:
@boocmp if we do the migration for desktop users as well then it would be affecting a lot of users. and migration would be setting fingerprinting option from `block restricted` to `default` for all users on android and desktop. 
IN that case, what if we don't do the migration at all and just change the option for android users only. 
CC @bridiver

Reply:
But if you change something in content settings on android it will be synced anyway with desktop (becasue content settings are syncable). I mean you should check your solution with sync enabled/disabled, if it doesn't break anything then it's OK.

Reply:
if you have sync enabled it's fine because any settings synced from android -> desktop will also be ambiguous and need to be reset